.github/workflows/integration.yaml

```diff
@@ -0,0 +1,81 @@
+name: Tests
+
+on:
+  pull_request:
+  schedule:
+    - cron: '0 8 * * TUE'
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: pip install tox
+
+    - name: Lint code
+      run: tox -vve lint
+
+  unit-test:
+    name: Unit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: pip install tox
+
+      - name: Run unit tests
+        run: tox -e unit
+
+      # Upload coverage report on failure
+      - name: Upload coverage artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: coverage
+          path: training-operator-cov_html
+        if: failure()
+
+  integration-test:
+    name: Integration
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+            provider: microk8s
+            channel: 1.21/stable
+
+      # https://github.com/charmed-kubernetes/actions-operator/issues/27
+      - name: Install charmcraft latest/candidate
+        run: sudo snap refresh charmcraft --channel latest/candidate
+
+      - name: Run integration tests
+        run: tox -e integration -- --model testing
+
+      # On failure, capture debugging resources
+      - name: Get all
```

Comment:
Another stylistic choice, but I think we could just name this step `Capture debugging resources on failure` and then skip the comment.